packages/pyright-internal/src/languageService/completionProvider.ts

```diff
@@ -2497,13 +2498,16 @@ export class CompletionProvider {
         });
     }
 
-    private _addNamedParametersToMap(type: FunctionType, paramMap: Map<string, string>) {
+    private _addNamedParametersToMap(type: FunctionType, names: Set<string>) {
         type.details.parameters.forEach((param) => {
-            if (param.name && !param.isNameSynthesized) {
+            if (isTypedKwargs(param) && param.type.category === TypeCategory.Class) {
```

Comment:
Isn't `param.type.category === TypeCategory.Class` guaranteed if `isTypedKwargs(param)` since `isTypedKwargs` checks `isClassInstance`?

Reply:
yes it is but typescript isn't smart enough to know that. Otherwise it shows a compiler error for using the details.